### PR TITLE
feat: update to ESM compatible version of CTF library

### PR DESCRIPTION
**Pull request by axe312ger**

### CODE REVIEW

**Comment by serhalp** [comment]
Thanks @axe312ger!

It looks like the tests are failing, e.g.

```
/home/circleci/project/packages/gatsby-source-contentful/src/__tests__/fetch-network-errors.js

  ● fetch-retry › request should fail after to many retries

    expect(received).toEqual(expected) // deep equality

    Expected: StringContaining "Fetching contentful data failed: ERR_BAD_RESPONSE 500 MockedContentfulError"
    Received: "Fetching contentful data failed: ERR_BAD_RESPONSE 500 500 MockedContentfulError Mocked message of Contentful error··
    ---
    Request ID: 123abc·
    Attempts: 3·
    ---
```

**Comment by axe312ger** [comment]
Yes, thanks. Just on it! :)

**Comment by axe312ger** [comment]
Had to do [772d895](https://github.com/gatsbyjs/gatsby/pull/39178/commits/772d89506f4a907273a0e67e2faa5d9e4084533d) and [8d93859](https://github.com/gatsbyjs/gatsby/pull/39178/commits/8d93859ddf58203ae8ffcd7939d6fa1860253d88) to make CI happy. Please extract, remove, ignore, merge as you like @Gatsby Team

File: .circleci/config.yml [hunk 1]
```diff
@@ -502,7 +502,7 @@ jobs:
   e2e_tests_adapters:
     <<: *e2e-executor
     docker:
-      - image: cypress/browsers:node-18.16.1-chrome-114.0.5735.133-1-ff-114.0.2-edge-114.0.1823.51-1
+      - image: cypress/browsers:node-18.17.19-chrome-114.0.5735.133-1-ff-114.0.2-edge-114.0.1823.51-1
```
Comment (pieh) [inline .circleci/config.yml:505]:
```suggestion
      - image: cypress/browsers:node-18.20.3-chrome-125.0.6422.141-1-ff-126.0.1-edge-125.0.2535.85-1
```

Per https://hub.docker.com/r/cypress/browsers/tags?name=node-18

File: .circleci/config.yml [hunk 2]
```diff
@@ -513,7 +513,7 @@ jobs:
   e2e_tests_adapters_monorepo:
     <<: *e2e-executor
     docker:
-      - image: cypress/browsers:node-18.16.1-chrome-114.0.5735.133-1-ff-114.0.2-edge-114.0.1823.51-1
+      - image: cypress/browsers:node-18.17.19-chrome-114.0.5735.133-1-ff-114.0.2-edge-114.0.1823.51-1
```
Comment (pieh) [inline .circleci/config.yml:516]:
```suggestion
      - image: cypress/browsers:node-18.20.3-chrome-125.0.6422.141-1-ff-126.0.1-edge-125.0.2535.85-1
```

same as above

**Review by pieh — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (axe312ger) [reply .circleci/config.yml:505]: thanks

**Review by pieh — APPROVED** [review]

**Pull request merged**